test/test_sync.cpp

```diff
@@ -6100,6 +6100,261 @@ TEST(Sync_Dictionary)
+        Mixed val = dict->get("hello");
+        CHECK_EQUAL(val.get_string(), "world");
+        val = dict->get("cnt");
+        CHECK_EQUAL(val.get_int(), 7);
+        val = dict->get("when");
+        CHECK_EQUAL(val.get<Timestamp>(), now);
+        CHECK_EQUAL(dict->get_list("list")->get(0).get_int(), 5);
+
+        // Erase dictionary element
+        dict->erase("cnt");
+        // Replace dictionary element
+        dict->insert("hello", "goodbye");
+
+        obj = table->get_object_with_primary_key(456);
+        auto list = obj.get_list_ptr<Mixed>(col_any);
+        // Check that values are replicated
+        CHECK_EQUAL(list->get(0).get_int(), 5);
+        CHECK_EQUAL(list->get(1).get_string(), "John");
+        // Replace list element
+        list->set(1, "Paul");
+        // Erase list element
+        list->remove(0);
+    });
+
+    session_2.wait_for_upload_complete_or_client_stopped();
+    session_1.wait_for_download_complete_or_client_stopped();
+
+    write_transaction_notifying_session(db_1, session_1, [&](WriteTransaction& tr) {
+        auto table = tr.get_table("class_Table");
+        auto col_any = table->get_column_key("any");
+        CHECK_EQUAL(table->size(), 2);
+
+        auto obj = table->get_object_with_primary_key(123);
+        auto dict = obj.get_dictionary(col_any);
+        CHECK_EQUAL(dict.size(), 3);
+
+        Mixed val = dict["hello"];
+        CHECK_EQUAL(val.get_string(), "goodbye");
+        val = dict.get("when");
+        CHECK_EQUAL(val.get<Timestamp>(), now);
+
+        // Dictionary clear
+        dict.clear();
+
+        obj = table->get_object_with_primary_key(456);
+        auto list = obj.get_list_ptr<Mixed>(col_any);
+        CHECK_EQUAL(list->size(), 1);
+        CHECK_EQUAL(list->get(0).get_string(), "Paul");
+        // List clear
+        list->clear();
+    });
+
+    session_1.wait_for_upload_complete_or_client_stopped();
+    session_2.wait_for_download_complete_or_client_stopped();
+
+    write_transaction_notifying_session(db_2, session_2, [&](WriteTransaction& tr) {
+        auto table = tr.get_table("class_Table");
+        auto col_any = table->get_column_key("any");
+
+        CHECK_EQUAL(table->size(), 2);
+
+        auto obj = table->get_object_with_primary_key(123);
+        auto dict = obj.get_dictionary(col_any);
+        CHECK_EQUAL(dict.size(), 0);
+
+        // Replace dictionary with list on property
+        obj.set_collection(col_any, CollectionType::List);
+
+        obj = table->get_object_with_primary_key(456);
+        auto list = obj.get_list<Mixed>(col_any);
+        CHECK_EQUAL(list.size(), 0);
+        // Replace list with dictionary on property
+        obj.set_collection(col_any, CollectionType::Dictionary);
+    });
+
+    session_2.wait_for_upload_complete_or_client_stopped();
+    session_1.wait_for_download_complete_or_client_stopped();
+
+    {
+        ReadTransaction read_1{db_1};
+        ReadTransaction read_2{db_2};
```

Comment:
we could check both realms have the same data, i.e,:
`
CHECK(compare_groups(rt_1, rt_2));
`

Reply:
:+1: